snakemake/executors.py

```diff
@@ -2115,3 +2128,759 @@ def run_wrapper(
+            "environment": self._generate_environment(),
+        }
+
+        # "timeout": string in seconds (3.5s) is not included (defaults to 7 days)
+        return pipeline
+
+    def run(self, job, callback=None, submit_callback=None, error_callback=None):
+
+        super()._run(job)
+
+        # https://cloud.google.com/life-sciences/docs/reference/rest/v2beta/projects.locations.pipelines
+        pipelines = self._api.projects().locations().pipelines()
+
+        # pipelines.run
+        # https://cloud.google.com/life-sciences/docs/reference/rest/v2beta/projects.locations.pipelines/run
+
+        labels = self._generate_pipeline_labels(job)
+        pipeline = self._generate_pipeline(job)
+
+        # The body of the request is a Pipeline and labels
+        body = {"pipeline": pipeline, "labels": labels}
+
+        # capabilities - this won't currently work (Singularity in Docker)
+        # We either need to add CAPS or run in privileged mode (ehh)
+        if job.needs_singularity and self.workflow.use_singularity:
+            logger.warning(
+                "Singularity requires additional capabilities that "
+                "aren't yet supported for standard Docker runs."
+            )
+
+        # location looks like: "projects/<project>/locations/<location>"
+        operation = pipelines.run(parent=self.location, body=body)
+
+        # 403 will result if no permission to use pipelines or project
+        result = self._retry_request(operation)
+
+        # The jobid is the last number of the full name
+        jobid = result["name"].split("/")[-1]
+
+        # Give some logging for how to get status
+        logger.info(
+            "Get status with:\n"
+            "gcloud config set project {project}\n"
+            "gcloud beta lifesciences operations describe {jobid}\n"
+            "gcloud beta lifesciences operations list".format(
+                project=self.project, jobid=jobid
+            )
+        )
+
+        self.active_jobs.append(
+            GoogleLifeSciencesJob(job, result["name"], jobid, callback, error_callback)
+        )
+
+    def _job_was_successful(self, status):
+        """based on a status response (a [pipeline].projects.locations.operations.get
+           debug print the list of events, return True if all return codes 0
+           and False otherwise (indication of failure). In that a nonzero exit
+           status is found, we also debug print it for the user.
+        """
+        success = True
+
+        # https://cloud.google.com/life-sciences/docs/reference/rest/v2beta/Event
+        for event in status["metadata"]["events"]:
+
+            logger.info(event["description"])
+
+            # Does it always result in fail for other failure reasons?
+            if "failed" in event:
+                success = False
+                action = event.get("failed")
+                logger.error("{}: {}".format(action["code"], action["cause"]))
```

Comment:
Please provide an example of what could be displayed here. Then, I'll let you know how we print it.

Reply:
Sure! See the first red line at the top - the FAILED_PRECONDITION is the content of `action["code"]` and then the cause follows. It's just that one little line.

![image](https://user-images.githubusercontent.com/814322/75285933-c0599100-57d4-11ea-8e2d-d7d3d24037f6.png)

Note that the rest of the message hasn't been updated for the changes just pushed to change info / error to debug.

Reply:
What I can do is finish up changes as you suggest here, then run it again and share the updated output.

Reply:
okay here is updated terminal output - you are correct that it's a lot more cleaner!

![image](https://user-images.githubusercontent.com/814322/75287596-e2a0de00-57d7-11ea-988b-4904a78a85c4.png)

I'm worried that there is sometimes meaningful information in the traceback that is printed, but this way is definitely cleaner looking, so I'm good to give it a try if you think this is best practice (notice that the job is still not exited so we might not see all of it yet).

Reply:
here is the entire thing (with the final error output)

![image](https://user-images.githubusercontent.com/814322/75290211-2813da00-57dd-11ea-9350-52f0cd1e0e1b.png)

Reply:
So, this failed_precondition line still seems to be superflous to me. People can always get the full detail with describe, can't they? @moschetti is there any information in the returned error dict that we cannot get from the describe command?